File: Semester3/Advanced Methods of Programming/Toy Language Interpreter JavaFx/Assignment8/src/View/TextMenu.java
package View;

import java.util.HashMap;
import java.util.Map;
import java.util.Scanner;

public class TextMenu {
    private Map<String, Command> commands;

    public TextMenu(){
        commands = new HashMap<>();
    }

    public void addCommand(Command c){
        commands.put(c.getKey(), c);
    }

    private void printMenu(){
        for(Command com : commands.values()){
            String line = String.format("%4s : %s", com.getKey(), com.getDescription());
            System.out.println(line);
        }
    }

    public void show(){
        Scanner scanner=new Scanner(System.in);
        while(true){
            printMenu();
            System.out.printf("Input the option: ");
            String key=scanner.nextLine();
            Command com=commands.get(key);
            if (com==null){
                System.out.println("Invalid Option");
                continue; }
            com.execute();
        }
    }
}

File: Semester3/Advanced Methods of Programming/Toy Language Interpreter C#/Assignement7/Assignement7/View/TextMenu.cs
using Assignement7.Collection;
using Assignement7.Model;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Assignement7.View
{
    class TextMenu
    {
        private MyIDictionary<string, Command> cmds;

        public TextMenu(MyIDictionary<string, Command> cmds) => this.cmds = cmds;

        public void AddCommand(Command cmd)
        {
            this.cmds.Put(cmd.GetKey(), cmd);
        }

        public void printMenu()
        {
            Console.WriteLine("Available commands: ");
            foreach (Command cmd in this.cmds.Values())
            {
                String line = String.Format("Command {0}: {1}", cmd.GetKey(), cmd.GetDescription());
                Console.WriteLine(line);
            }
        }

        public List<string> getCommandList()
        {
            List<string> l = new List<String>();
            foreach (Command cmd in this.cmds.Values())
                l.Add(cmd.GetDescription());
            return l;
        }

        public void show()
        {
            while (true)
            {
                printMenu();
                Console.WriteLine("Input the command: ");
                Command cmd = cmds.Get(Console.ReadLine());
                if (cmd == null)
                {
                    Console.WriteLine("Invalid command");
                    continue;
                }
                cmd.Execute();
            }
        }
    }
}
